python/tests/dexjit_test.py

```diff
@@ -0,0 +1,128 @@
+                                   lambda: (1.0, rn(10), rn(10)))
+  test_array_select_lt = lax_test(lambda i, x, y: lax.select(i < 0.0, x, y),
+                                  lambda: (rn(10), rn(10), rn(10)))
+
+  test_squeeze_none = lax_test(lambda x: lax.squeeze(x, [ ]), lambda: (rn(10, 10),))
+  test_squeeze_one = lax_test(lambda x: lax.squeeze(x, [1]), lambda: (rn(10, 1, 10),))
+  test_squeeze_two = lax_test(lambda x: lax.squeeze(x, [0, 2]), lambda: (rn(1, 10, 1),))
+  test_squeeze_all = lax_test(lambda x: lax.squeeze(x, [0, 1]), lambda: (rn(1, 1),))
+
+  test_slice_1d = lax_test(lambda x: lax.slice(x, [2], [5], None), lambda: (rn(10),))
+  test_slice_3d = lax_test(lambda x: lax.slice(x, [2, 0, 0], [5, 10, 2], None), lambda: (rn(10, 10, 2),))
+
+  test_concat_uniform = lax_test(partial(lax.concatenate, dimension=0),
+                                 lambda: ([rn(4, 2) for _ in range(3)],))
+  test_concat_ragged = lax_test(partial(lax.concatenate, dimension=0),
+                                lambda: ([rn(1, 2, 4), rn(5, 2, 4), rn(2, 2, 4)],))
+
+  test_dot_general_matmul = lax_test(partial(lax.dot_general, dimension_numbers=(((1,), (0,)), ((), ()))),
+                                     lambda: (rn(4, 8), rn(8, 16)))
+  test_dot_general_matvec = lax_test(partial(lax.dot_general, dimension_numbers=(((1,), (0,)), ((), ()))),
+                                     lambda: (rn(4, 8), rn(8)))
+
+  def test_canonicalize_dtype(self):
+    c = np.arange(5, dtype=np.float64)
+    f = lambda x: x * c
+    x = np.ones(5, dtype=np.float64)
+    dy = dexjit(f)(x)
+    jy = jax.jit(f)(x)
+    np.testing.assert_allclose(dy, jy)
+    self.assertEqual(dy.dtype, jy.dtype)
+
+  def test_64_bit(self):
+    def f(x):
+      return x * x + 4.0
+    x = np.float64(2.0)
+    with enable_x64():
+      np.testing.assert_allclose(dexjit(f)(x), 8.0)
+
+  def test_jit(self):
+    g = dexjit(lambda y: y * y)
+    f = jax.jit(lambda x: x * g(x))
+    x = jnp.arange(5)
+    np.testing.assert_allclose(f(x), x * x * x)
+
+  def test_jit_two_outputs(self):
+    f = jax.jit(dexjit(lambda x, y: (x + 1, y * 4)))
+    x = jnp.arange(10, dtype=np.float32)
+    y = 5.0
+    for l, r in zip(f(x, y), (x + 1, y * 4)):
+      np.testing.assert_allclose(l, r)
+
+
+def check_broadcasting_pointwise(prim, full=False):
+  setattr(JAX2DexTest, 'test_' + prim.__name__,
+          lax_test(prim, lambda: (rn(10, 10), rn(10, 10))))
+  if full:
+    setattr(JAX2DexTest, 'test_' + prim.__name__ + '_expand',
+            lax_test(prim, lambda: (rn(10, 10), rn(10, 1))))
+    setattr(JAX2DexTest, 'test_' + prim.__name__ + '_scalar',
+            lax_test(prim, lambda: (rn(10, 10), rn())))
+check_broadcasting_pointwise(lax.add, full=True)
+check_broadcasting_pointwise(lax.mul)
+check_broadcasting_pointwise(lax.sub)
+check_broadcasting_pointwise(lax.div)
+check_broadcasting_pointwise(lax.max)
+
+
+if __name__ == "__main__":
+  unittest.main()
```

Comment:
Any particular reason to split off this file?

Reply:
Theme -- calling Dex from Jax and transpiling Jax to Dex register as quite different interop mechanisms (in particular at quite different stages of completeness), so it seems sensible to have their tests live in different files (especially as the amount of code therein has been growing).